apps/embed-iframe/src/LoginButtonComponent.tsx

```diff
@@ -0,0 +1,62 @@
+import { Box, Button, Flex, Heading } from "@chakra-ui/react";
+import { type TypeOfLogin } from "@trilitech-umami/umami-embed/types";
+
+import { FacebookLogoIcon, GoogleLogoIcon, RedditLogoIcon, TwitterLogoIcon } from "./assets/icons";
+
+export const LoginButtonComponent: React.FC<{
+  onClick: () => void;
+  loginType: TypeOfLogin;
+}> = ({ onClick, loginType }) => (
+  <Button position="relative" width="100%" onClick={onClick} size="lg" borderRadius="36px" padding="0px">
+    <Box position="absolute" left="0px" top="50%" transform="translateY(-50%)">
+      <LogoIconWithBackground loginType={loginType} />
+    </Box>
+    <Heading textAlign="center" flex="1" fontSize="14px" lineHeight="18px">
+        {buttonLabel(loginType)}
+     </Heading>
+  </Button>
+);
+
+const LogoIconWithBackground: React.FC<{
+  loginType: TypeOfLogin;
+}> = ({ loginType }) => (
+    <Flex
+      alignItems="center"
+      justifyContent="center"
+      width="36px"
+      height="36px"
+      borderRadius="50%"
+      backgroundColor="white"
+      margin="6px"
+    >
+      <LogoIcon loginType={loginType} />
+    </Flex>
+  );
+
+const LogoIcon: React.FC<{
+  loginType: TypeOfLogin;
+}> = ({ loginType }) => {
+  switch (loginType) {
+    case "facebook":
+      return <FacebookLogoIcon position="absolute" />;
+    case "google":
+      return <GoogleLogoIcon position="absolute" />;
+    case "twitter":
+      return <TwitterLogoIcon position="absolute" />;
+    case "reddit":
+      return <RedditLogoIcon position="absolute" />;
+  }
+};
+
+const buttonLabel = (loginType: TypeOfLogin) => {
+  switch (loginType) {
+    case "facebook":
+      return "Facebook";
+    case "google":
+      return "Google";
+    case "twitter":
+      return "X";
+    default:
```

Comment:
it's better not to use the default. because when we add the email provider, this won't notify you that you forgot to update it. eslint will check the switch-case exhaustiveness

Reply:
Good catch! Forgot to rewrite this bit.